messages/src/lib.rs

```diff
@@ -426,16 +431,144 @@ impl ModuleId {
         }
     }
 
+    /// Convenience method to build a `ModuleId` that selects no transceivers on
+    /// the given FPGA.
+    pub const fn empty(fpga_id: u8) -> Self {
+        Self {
+            fpga_id,
+            ports: PortMask::empty(),
+        }
+    }
+
     /// Return `true` if the provided index is contained in set of addressed
     /// modules.
     pub const fn contains(&self, ix: u8) -> bool {
         self.ports.contains(ix)
     }
 }
 
+/// The MAC address allocation for the target system.
+#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize, SerializedSize)]
+pub struct MacAddrs {
+    // The first valid MAC address for the system.
+    base_mac: [u8; 6],
+    // The number of available addresses following `base_mac`.
+    count: u16,
+    // The stride pattern used to compute the next available MAC address.
+    //
+    // This is used in situations in which subsystems may make assumptions
+    // about spacing between addresses. At the time of writing, this applies to
+    // the Chelsio T6 NICs.
+    stride: u8,
+}
+
+impl MacAddrs {
+    /// Create a new MAC address range.
+    ///
+    /// This will do some basic sanity checks on the provided data to make sure
+    /// that the full range is valid. Specifically, we check that the base MAC
+    /// is not a multicast address, that all addresses in the range share the
+    /// same OUI, and that count/stride are nonzero.
+    pub const fn new(base_mac: [u8; 6], count: u16, stride: u8) -> Option<Self> {
```

Comment:
Since this now returns an `Option`, can you add a `BadMacAddrBlock` to the [`enum Error`](https://github.com/oxidecomputer/transceiver-control/blob/main/messages/src/lib.rs#L43) above?  This will allow Hubris to report a meaningful error if it's got bad data (instead of unwrapping it).

Reply:
I would vote for including the bad data as "what was the bad address" is the first question one would ask I'd think. Closely followed by "how did we generate it". Being able to jump right into the next step of debugging seems valuable.

Reply:
Wait I think I misunderstood your suggestion, @Aaron-Hartwig. You mean include the exact base / stride / count in the error, rather than a description of the error?

Reply:
I think it should be the bad data (base + stride + count) _with_ a description (?) rather than just a description on its own.